Create A Blockchain With Python

corpsgeek opened the pull request:
### Article checklist

#### Pre-submission checks
> Please ensure that you have only one open topic suggestion (issue) + in-review article (linked pull request) at a time. This will ensure that we complete the review process in a timely manner from inception to publishing

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
> Remove the backticks and add the issue number below to link and close your Topic Suggestion (issue) when your article has been published (PR has been merged). See this [video](https://youtu.be/TKJ4RdhyB5Y?t=278) for more details.

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

corpsgeek commented:
This PR is linked to https://github.com/section-engineering-education/engineering-education/issues/3654

ahmadmardeni1 commented:
You have too many files. Please delete them @corpsgeek

ahmadmardeni1 commented:
plagiarism check done ✅

corpsgeek commented:
Done @ahmadmardeni1 , @adrianmurage

corpsgeek commented:
@adrianmurage how's the review process progressing?

corpsgeek commented:
@hectorkambow is my reviewer still active??? He/she seems MIA.

adrianmurage commented:
@corpsgeek I am still here. I just got a little tied up with school work
I'll make the review ASAP
Thank you for your patience

adrianmurage commented:
@corpsgeek 
Great job with the article 
Small note: Other than the hero image, the remaining attached images seem to not be in use. Kindly delete them

adrianmurage reviewed (changes requested):
@corpsgeek
Great job with the article
Small note: Other than the hero image, the remaining attached images seem to not be in use. Kindly delete them

corpsgeek commented:
Hi @adrianmurage the images are actually relevant, I updated the links in the content to locate the links properly and I also think I meesed up the fetching of updates a bit, would be glad if you can restore the changes you made while keeping the links i updated intact. Thank you

adrianmurage commented:
@corpsgeek 
You seem to have mixed up the files for two different articles altogether 
I had concluded the review for the article titled something in the lines of "getting started with blockchain development"
And now I see that the current article is on "building a blockchain with python"

Kindly clarify what article is relevant to this pr and make sure that only files related to the relevant article are present in the pr

corpsgeek commented:
@adrianmurage please check your commit on October 1 i fetched what was in the PR and based on your commit, it doesn’t seem to be my article. Please check your end and try to rectify cause I’m getting really lost and confused amongst this.

corpsgeek reviewed (commented):
I deleted the index.md from your commit not related to mine.

corpsgeek commented:
@adrianmurage thats the changes you made and commuted last and it wasn’t my article. 

Cc: @hectorkambow

adrianmurage commented:
@corpsgeek I was reviewing the file
It seems your pr had files that were unrelated to it hence the confusion

I'll proceed to review the correct article now

adrianmurage commented on content/articles/create-a-blockchain-in-python/index.md:
````diff
@@ -0,0 +1,506 @@
+```
+
+### Checking The Validity of The Blockchain
+In this step, we create a function that checks if the entire blockchain is valid. This step is crucial in maintaining the integrity of our blockchain to ensure that none of our block is corrupt because if the data in a block changes, the hash changes and the link between each block in the chain is broken. This process is well explained in my previous [article](https://www.section.io/engineering-e ...
````
```suggestion
In this step, we create a function that checks if the entire blockchain is valid.

This step is crucial in maintaining the integrity of our blockchain to ensure that none of our blocks is corrupt because if the data in a block changes, the hash changes, and the link between each block in the chain is broken. This process is well explained in my previous [article](https://www.section.io/engineering-education/an-introduction-to-blockchain-architecture/#mining-and-proof-of-work).
```

adrianmurage reviewed (changes requested):
Great job @corpsgeek 

I left a change to be made. Also, the headings should be h3 and above.
Looking forward to seeing more of your work!

corpsgeek commented:
Should the revision label still be there? @adrianmurage
